cpp/tests/utilities/debug_utilities.cu

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) 2019-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Do NOT include any header that directly or indirectly depends on gtest.
```

Comment:
Does GTest define any macros that we can check for to raise an error?

Reply:
Good idea. There is a macro `GTEST_INCLUDE_GTEST_GTEST_H_`  that is currently used in `cudf_gtest.hpp` to detect gtest.

Reply:
The macro check was undone. Is this no longer needed then?

Reply:
This PR now falls back to a normal refactor PR that extracts `cudf::test::print` into a separate compilation unit.